packages/slate/test/interfaces/Text/decorations/collapse.js

```diff
@@ -0,0 +1,85 @@
+import { Text } from 'slate'
+
+export const input = [
+  {
+    anchor: {
+      path: [0],
+      offset: 1,
+    },
+    focus: {
+      path: [0],
+      offset: 2,
+    },
+    decoration1: 'decoration1',
+  },
+  {
+    anchor: {
+      path: [0],
+      offset: 2,
+    },
+    focus: {
+      path: [0],
+      offset: 2,
+    },
+    decoration2: 'decoration2',
+  },
+  {
+    anchor: {
+      path: [0],
+      offset: 2,
+    },
+    focus: {
+      path: [0],
+      offset: 3,
+    },
+    decoration3: 'decoration3',
+  },
+  {
+    anchor: {
+      path: [0],
+      offset: 4,
+    },
+    focus: {
+      path: [0],
+      offset: 4,
+    },
+    decoration4: 'decoration4',
+  },
+]
+
+export const test = decorations => {
+  return Text.decorations({ text: 'abcd', mark: 'mark' }, decorations)
+}
+
+export const output = [
+  {
+    text: 'a',
+    mark: 'mark',
+  },
+  {
+    text: 'b',
+    mark: 'mark',
+    decoration1: 'decoration1',
+  },
+  {
+    text: '',
+    mark: 'mark',
+    decoration1: 'decoration1',
```

Comment:
wouldn't without decoration1 here be the more natural result? since decoration1 indeed just care the text 'b', not the zero-length item after that?

Reply:
This is additional decoration between two character segments, so it should contains all decorations overlapped in this point. This will be useful, If you want to control multi-cursor overlap (for example). In this case you need to know, where they overlap in order to show this overlap in view.